sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/models/RuleFilter.java

```diff
@@ -27,4 +27,11 @@
         // This is intentionally left blank. This constructor exists
         // only to prevent external assemblies inheriting from it.
     }
+
+    @Override
+    public abstract boolean equals(Object obj);
+
+    @Override
+    public abstract int hashCode();
```

Comment:
Why is this being done?

Reply:
The idea behind adding this was to enforce any new implementation of RuleFilter to always implement equals and hashcode methods. This is not a hard requirement, but this establishes a contract which is good to have

Reply:
I would preferably not have this be introduced, just because it is a breaking change in the API.

Reply:
Ah even without these two methods, introduing equals and hashcode methods in the other two classes already causes a breaking API change, which is why I thought, we are already doing this, so lets formalize this by introducing this into the base class

Reply:
Can I ask why this is considered a breaking API change?  These two methods are part of the base Java Object Model.  These are defined in Object.java.  All this is requesting it to override these existing methods to implementations that are more appropriate for the relative classes.

I presume there is another specification in play here, which would determine these breaking changes?

Reply:
Removed this from the current PR, will send out another one with these changes so that we can continue the discussion on inclusion of these changes